### ISSUE TITLE
How to read partial data from the remote files using downloadTo API? 

### ISSUE THREAD

**Issue by aditya81070** [patrickjuchli/basic-ftp#252]
**Describe the bug**
This is not a bug but rather a question related to working of download API. 

I have already checked https://github.com/patrickjuchli/basic-ftp/issues/58 and understand how can I use passThrough or writeStreams to download the files or simultaneously. I couldn't find any example for reading partial data from a file. 

**Example code**
A short and standalone code example that demonstrates the issue. Don't copy-paste a few lines from your project, it makes it difficult to rule out issues in other parts of your code.
```
  const client = new Client();
  await client.access();
  const files = await client.list('/out');
  const readStream = new Stream.PassThrough();
  await client.downloadTo(transform, `/out/${selectedFile.name}`);
 // how to stop the above operation so it doesn't read complete file. 
```

**Console output**
Use `client.ftp.verbose = true` to log every communication detail. Copy-paste the output of your console here. Make sure to redact IPs, URLs or passwords.

**Which version of Node.js are you using?**
v21.2.0

**Which version of basic-ftp are you using?**
^5.0.5

**Additional context**
Add any other context about the problem here. What FTP server are you using? What environment are you running in?

**Comment by paulish** [comment]
Yes, need this feature too. I see that I can set the start byte but I see no way to set the end byte as it is possible in fs.createReadStream or s3.GetObject api or mongo GridFS bucket.openDownloadStream api. 
I need this feature to properly implement Range http header support for our file service rest api with file/ftp/s3/mongo/... adapters.

I suppose it is possible to somehow wrap destination to stop reading but have not tried yet. At any case it is better to implement in Reader than hacking with Writer.

**Comment by patrickjuchli** [comment]
Some servers support the [RANG command](https://www.ietf.org/archive/id/draft-bryan-ftp-range-03.html) to do that. I'm not sure when and if this library will implement this because it introduces state that might have other side-effects that need time to test/debug.

In the meantime, you might be able to do it yourself like so, haven't tested:

```js
await client.send(`RANG ${fromByte} ${toByte}`)
await client.downloadTo("localfile.txt", "remotefile.txt")
```

The download command has to immediately follow the RANG command. This command sets state on the FTP server and I'm not sure when exactly when it is reset. To reset manually, according to the spec, you can do this:

```js
await client.send("RANG 1 0")
```

Note that not all servers support this command. Check support using `client.features()`. Or use `client.sendIgnoringError` to ignore any error and download the whole file if must be.
